tests/python/pants_test/util/test_objects.py

```diff
@@ -352,20 +364,27 @@ class MultipleSameNameWithType(datatype([
     expected_msg = "Encountered duplicate field name: 'field_a'"
     self.assertEqual(str(cm.exception), str(expected_msg))
 
+    with self.assertRaises(DatatypeHelperConstructionError) as cm:
+      class InvalidTypeSpec(datatype([('a_field', 2)])): pass
+    expected_msg = (
+      "type spec for field 'a_field' was not a type or TypeConstraint: "
+      "was 2 (type 'int').")
+    self.assertEqual(str(cm.exception), str(expected_msg))
+
   def test_instance_construction_by_repr(self):
     some_val = SomeTypedDatatype(3)
     self.assertEqual(3, some_val.val)
     self.assertEqual(repr(some_val), "SomeTypedDatatype(val=3)")
     self.assertEqual(str(some_val), "SomeTypedDatatype(val<=int>=3)")
 
-    some_object = YetAnotherNamedTypedDatatype(str('asdf'), 45)
+    some_object = WithExplicitTypeConstraint(str('asdf'), 45)
     self.assertEqual(some_object.a_string, 'asdf')
     self.assertEqual(some_object.an_int, 45)
     self.assertEqual(repr(some_object),
-                     "YetAnotherNamedTypedDatatype(a_string='asdf', an_int=45)")
+                     "WithExplicitTypeConstraint(a_string='asdf', an_int=45)")
     self.assertEqual(
       str(some_object),
-      str("YetAnotherNamedTypedDatatype(a_string<=str>=asdf, an_int<=int>=45)"))
+      str("WithExplicitTypeConstraint(a_string<=str>=asdf, an_int<=int>=45)"))
```

Comment:
I see that there is precedent in this file, but why `str("...")`? This is redundant under both python2 and python3.

Reply:
I vaguely described the reasoning I thought there might be for this, then tried just removing the `str()` calls, and everything worked. No clue what issue I was having before.